src/pudl/analysis/demand_mapping.py

```diff
@@ -605,77 +605,90 @@ def allocate_and_aggregate(
+def categorize_eia_code(eia_codes, ba_ids, util_ids, priority="balancing_authority"):
     """
-    Categorize EIA Codes in FERC 714 as BA or Utility IDs.
+    Categorize FERC 714 ``eia_codes`` as either balancing authority or utility IDs.
 
     Most FERC 714 respondent IDs are associated with an ``eia_code`` which refers to
-    either a ``balancing_authority_id_eia`` or a ``utility_id_eia`` but no indication is
-    given as to which type of ID each one is. This is further complicated by the fact
+    either a ``balancing_authority_id_eia`` or a ``utility_id_eia`` but no indication
+    as to which type of ID each one is. This is further complicated by the fact
     that EIA uses the same numerical ID to refer to the same entity in most but not all
     cases, when that entity acts as both a utility and as a balancing authority.
 
-    Given the nature of the FERC 714 hourly demand dataset, this function assumes that
-    if the ``eia_code`` appears in the EIA 861 Balancing Authority table, that it should
-    be labeled ``balancing_authority``. If the ``eia_code`` appears only in the EIA 860
-    Utility table, then it is labeled ``utility``. These labels are put in a new column
-    named ``respondent_type``. If the planning area's ``eia_code`` does not appear in
-    either of those tables, then ``respondent_type`` is set to NA.
+    This function associates a ``respondent_type`` of ``utility``,
+    ``balancing_authority`` or ``pandas.NA`` with each input ``eia_code`` using the
+    following rules:
+    * If a ``eia_code`` appears only in ``util_ids`` the ``respondent_type`` will be
+    ``utility``.
+    * If ``eia_code`` appears only in ``ba_ids`` the ``respondent_type`` will be
+    assigned ``balancing_authority``.
+    * If ``eia_code`` appears in neither set of IDs, ``respondent_type`` will be
+    assigned ``pandas.NA``.
+    * If ``eia_code`` appears in both sets of IDs, then whichever ``respondent_type``
+    has been selected with the ``priority`` flag will be assigned.
+
+    Note that the vast majority of ``balancing_authority_id_eia`` values also show up
+    as ``utility_id_eia`` values, but only a small subset of the ``utility_id_eia``
+    values are associated with balancing authorities. If you use
+    ``priority="utility"`` you should probably also be specifically compiling the list
+    of Utility IDs because you know they should take precedence. If you use utility
+    priority with all utility IDs
 
     Args:
-        rids_ferc714 (pandas.DataFrame): The FERC 714 ``respondent_id`` table.
-        utils_eia860 (pandas.DataFrame): The EIA 860 Utilities output table.
-        ba_eia861 (pandas.DataFrame): The EIA 861 Balancing Authority table.
+        eia_codes (ordered collection): A dataframe based on the respondent_id_ferc714
+            table, having at least the two columns: ``respondent_id_ferc714`` and
+            ``eia_code``. Other columns may be present.
```

Comment:
Is this supposed to be an ordered list of a dataframe? It looks like a list or dict of just codes from the way it is used below. I assume the description here is old/wrong. Please clarify.

Reply:
Oh, actually this is just totally wrong and left over from the previous version. I will fix it.